client/src/components/Applications/AppForm/AppFormUtils.js

```diff
@@ -47,3 +49,52 @@ export const createMaxRunTimeRegex = maxRunTime => {
   });
   return `${regStr}|${upperReg}`;
 };
+
+/**
+ * Get validator for a node count of a queue
+ *
+ * @function
+ * @param {Object} queue
+ * @returns {Yup.number()} min/max validation of node count
+ */
+export const getNodeCountValidation = queue => {
+  // all queues have a min node count of 1 except for the normal queue.
+  const min = queue.name === 'normal' ? 3 : 1;
+  return Yup.number()
+    .min(
+      min,
+      `Node Count must be greater than or equal to ${min} for the ${queue.name} queue`
+    )
+    .max(
+      queue.maxNodes,
+      `Node Count must be less than or equal to ${queue.maxNodes} for the ${queue.name} queue`
+    );
+};
+
+/**
+ * Get validator for processors on each node
+ *
+ * @function
+ * @param {Object} queue
+ * @returns {Yup.number()} min/max validation of maxProcessorsPerNode
+ */
+export const getProcessorsOnEachNodeValidation = queue => {
+  if (queue.maxProcessorsPerNode === -1) {
+    return Yup.number();
+  }
+  return Yup.number()
+    .min(1)
+    .max(queue.maxProcessorsPerNode / queue.maxNodes);
+};
+
+export const getQueueValidation = (queue, app) => {
+  return Yup.string()
+    .required('Required')
+    .oneOf(app.exec_sys.queues.map(q => q.name))
+    .test(
+      'is-not-serial-job-using-normal-queue',
+      'The normal queue does not support serial apps (i.e. Node Count set to 1).',
+      (value, context) =>
+        queue.name !== 'normal' || app.definition.parallelism !== 'SERIAL'
+    );
```

Comment:
Can you explain how this works/what this does? Not sure I understand

Reply:
For SERIAL apps, they are single-node job and we don't show/provide the node-count input for users (and they can't see the validation error for number of nodes).  So I'm adding a validation check here for the case of a serial job using the normal queue (which doesn't allow single node jobs).

From discussing this, it feels like we should possibly instead:
A) remove `normal` queue from dropdown if `SERIAL` app
B) maybe refactor the check (**or just actually remove it if we do (A**) ) from
```
queue.name !== 'normal' || app.definition.parallelism !== 'SERIAL'
```
to
```
!(queue.name === 'normal' && app.definition.parallelism === 'SERIAL')
```
what do you think?

Reply:
In this case, I think A) should suffice, and also maintains that the `oneOf(app.exec_sys.queues.map(q => q.name))` check is still in place

Reply:
done 👍